group/assignment6/RearrangeCars.cpp

```diff
@@ -139,3 +139,34 @@ void RearrangeCars::generateSequenceOfMovesWithConstraints(const vector<int> &en
 			break;
 	}
 }
+
+void RearrangeCars::bruteForceAllSequence(vector<int>& currentStates, const vector<int>& endStates, vector<int>& carPosition, int emptySlotId, vector<Move>& sequenceOfMoves, vector<vector<Move>>& sequencesOfMoves, set<vector<int>>& usedPositions) {
+	int numOfCarsOnDisiredPosition = numberOfCarsOnDesiredPosition(carPosition, endStates);
+	if (numOfCarsOnDisiredPosition == carPosition.size()) {
+		sequencesOfMoves.push_back(sequenceOfMoves);
+		return;
+	}
+	for (int currCarSlotId = 0; currCarSlotId < carPosition.size(); currCarSlotId++) {
+		if (currCarSlotId != emptySlotId) {

```

Comment:
I think we should add another check here - _if (currCarSlotId != emptySlotId && validMove(carId, emptySlotId, constraints))_, and then this solution also supports constraints

Reply:
Yeah, it's easy and add functionality to solution